internal/error.go

```diff
@@ -0,0 +1,80 @@
+package fleetlock
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// ErrorKind is a error type identifier.
+//
+// It is designed to be tracked in Zincati metrics.
+// Fleetlock specification states that this MUST have a bounded/small cardinality.
+type ErrorKind string
+
+// List of ErrorKinds
+const (
+	ErrorMethodNotAllowed ErrorKind = "method_not_allowed"
+	ErrorMissingHeader    ErrorKind = "missing_header"
+	ErrorDecodingRequest  ErrorKind = "failed_decoding_request"
+	ErrorInternal         ErrorKind = "internal_error"
+	ErrorLocked           ErrorKind = "locked"
+)
+
+// Error holds data used for http responses in case of error.
+type Error struct {
```

Comment:
Actually, it will trigger a Golint error:

>  type name will be used as fleetlock.FleetlockError by other packages, and that stutters; consider calling this Error

And as golint is enforced in Github actions, it will make CI fail.
Do you prefer:
- using `FleetlockError` and relaxing CI (golint [do not enable suppressing a specific warning](https://github.com/golang/lint#purpose)),
- keeping `Error`(standard library `error` is in lowercase)
- using another name? `ErrorResponse` ?

Reply:
How about something like `Reply` then? The fact the protocol only happens to recommend a JSON format for "error" responses today seems like a temporary detail to me and I think its more readable if errors relate to actual Go errors.

Reply:
I like it!
For success replies, do you want to continue using a plain-text message or do you want to also use a json reply? Currently, the specification only states that it must reply with a 200 status code, and airlock replies with no body at all (and zincati do not use the output).

Reply:
Yeah, definitely like the idea of all the replies being JSON formatted messages. Although maybe for now we just plan for it in the future so this PR is just about matching the spec.

Reply:
Ok!